src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -242,6 +244,19 @@ private ApiEvent InitializeApiEvent(string accountId)
             apiEvent.TokenType = AuthenticationRequestParameters.AuthenticationScheme.TelemetryTokenType;
             apiEvent.AssertionType = GetAssertionType();
 
+            // Check if ExtraQueryParameters contains caller-sdk-id and caller-sdk-ver
+            if (AuthenticationRequestParameters.ExtraQueryParameters.TryGetValue("caller-sdk-id", out string callerSdkId))
```

Comment:
Pls keep the "Extra query params" way, in addition to `WithClientName` and `WithClientVersion`. Some SDKs like Id.Web create just 1 CCA object, for use with various APIs. So we need the ability of configuring "caller SDK" on a per request basis. 

CC @localden @rayluo